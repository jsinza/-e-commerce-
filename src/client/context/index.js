import React, { createContext, useReducer } from 'react';
import { reducer, INITIAL_STATE } from '../reducers';
import { fetchFailure, fetchItemDetailStart, fetchItemDetailSuccess, fetchItemsStart, fetchItemsSuccess } from '../actions';
import * as itemService from '../services';

const StoreContext = createContext();

const StoreProvider = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, INITIAL_STATE);
  const value = { state, dispatch };

  return (
    <StoreContext.Provider value={value}>
      {children}
    </StoreContext.Provider>
  );
};

const useStore = () => {
  const context = React.useContext(StoreContext);
  if (context === undefined) {
    throw new Error('useCount must be used within a CountProvider');
  }
  return context;
};

const searchItems = async (dispatch, query) => {
  dispatch(fetchItemsStart(query));
  try {
    const { data } = await itemService.searchItems(query);
    dispatch(fetchItemsSuccess(data));
  } catch ({ message }) {
    dispatch(fetchFailure(message));
  }
};

const getItem = async (dispatch, id) => {
  dispatch(fetchItemDetailStart());
  try {
    const { data } = await itemService.getItem(id);
    dispatch(fetchItemDetailSuccess(data));
  } catch ({ message }) {
    dispatch(fetchFailure(message));
  }
};

export { useStore, StoreProvider, searchItems, getItem };
