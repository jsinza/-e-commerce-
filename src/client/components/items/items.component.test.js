import React from 'react';
import { shallow } from 'enzyme';
import 'babel-polyfill';
import Items from './items.component';

import * as context from '../../context';

jest.mock('../../context');

const ITEMS = [
  {
    'id': 'MLA680573238',
    'title': 'Zapato Vulcano',
    'price': {
      'currency': 'ARS',
      'amount': 2199,
      'decimals': 0,
    },
    'picture': 'http://http2.mlstatic.com/D_645465-MLA31037702912_062019-O.jpg',
    'condition': 'new',
    'free_shipping': false,
  }];

describe('Items Component', () => {

  let wrapper;

  it('should render Items Component', () => {
    context.useStore.mockReturnValue({ state: { items: ITEMS, isLoading: false } });
    wrapper = shallow(<Items />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the items when you have already loaded the data', () => {
    context.useStore.mockReturnValue({ state: { items: ITEMS, isLoading: false } });
    wrapper = shallow(<Items />);
    expect(wrapper.find("[className='items']")).toHaveLength(1);
  });

  it('should not show the items if the data has not been loaded', () => {
    context.useStore.mockReturnValue({ state: { items: ITEMS, isLoading: true } });
    wrapper = shallow(<Items />);
    expect(wrapper.find("[className='items']")).toHaveLength(0);
  });

});
