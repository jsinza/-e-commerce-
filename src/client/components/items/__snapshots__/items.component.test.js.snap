// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Items Component should render Items Component 1`] = `ShallowWrapper {}`;
